src/include/lean/lean.h

```diff
@@ -570,26 +623,77 @@ static inline size_t lean_ctor_get_usize(b_lean_obj_arg o, unsigned i) {
     return *((size_t*)(lean_ctor_obj_cptr(o) + i));
 }
 
+/*
+Given a ctor-object, returns a scalar parameter of type `uint8_t` at a given offset. 
+
+The offset does *not* represent an index into the list of parameters of an inductive 
+type constructor, but is instead only a byte offset. To access a specific constructor 
+parameter, construct the correct byte offset as follows. 
+
+All scalar parameters of a ctor-object are represented in a contiguous block following
+the non-scalar parameters. Thus, the base offset required to access a scalar parameter
+of a ctor-object `o` is `lean_ctor_num_objs(o) * sizeof(void*)` (cf. the documentation 
+for `lean_ctor_num_objs` for more details). Then, to locate the desired scalar parameter, 
+add the sizes of  all scalar parameters preceeding the target parameter. For example, 
+let `o` be a value of the following structure in Lean:
+
+```
+structure S where
+    a : String
+    b : Bool
+    c : UInt64
+    e : Array Nat
+    d : UInt8
+```
+
+Then 
+* the base offset is `base_offset := 3 * sizeof(void*)`.
+* as `b` is the first scalar parameter, its offset is exactly the `base_offset`.
+* the offset for `c` is `base_offset + sizeof(uint8_t)`, where `sizeof(uint8_t)` 
+  is the number of bytes used by the preceeding parameter `b`. 
```

Comment:
```suggestion
  is the number of bytes used by the preceding parameter `b`. 
```

Also, where should readers go to find out how much space is used by `Bool` and thus `b`?

Reply:
https://lean-lang.org/lean4/doc/dev/ffi.html#translating-types-from-lean-to-c explains that `Bool` is represented as a `uint8_t`.